src/evidently/EvidentlyManager.ts

```diff
@@ -0,0 +1,261 @@
+            try {
+                const jsonRequest = sessionStorage.getItem(REQUEST_CACHE_KEY);
+                return JSON.parse(
+                    jsonRequest as string
+                ) as InitializeFeaturesRequest;
+            } catch (e) {
+                // Ignore
+            }
+        }
+        return undefined;
+    }
+
+    private updateLastRequest(request: InitializeFeaturesRequest) {
+        this.lastRequest = request;
+        if (this.useSessionStorage()) {
+            sessionStorage.setItem(REQUEST_CACHE_KEY, JSON.stringify(request));
+        }
+    }
+
+    private contextMatches(c1?: ContextType, c2?: ContextType): boolean {
+        // Note: Will return false if context has the same values but in a different order,
+        //  we ignore this edge case to avoid complex equality checking
+        return JSON.stringify(c1 || {}) === JSON.stringify(c2 || {});
+    }
+
+    private addEvaluationToEvents(evaluation: EvaluationResult) {
+        const evidentlySessionAttributes: {
+            [key: string]: string;
+        } = {
+            [evaluation.feature]: evaluation.variation
+        };
+        this.eventCache.addEvidentlyAttributes(evidentlySessionAttributes);
+    }
+
+    private getDefaultEntityId(): string {
+        const userDetails = this.eventCache.getUserDetails();
+        return (userDetails.userId === NIL_UUID
+            ? userDetails.sessionId
+            : userDetails.userId) as string;
+    }
+
+    private async fetchAPIEvaluations(
+        request: InitializeFeaturesRequest
+    ): Promise<EvaluationResults> {
+        const entityId = request.entityId || this.getDefaultEntityId();
+        const context = request.context
+            ? JSON.stringify(request.context)
+            : undefined;
+        const apiRequest: BatchEvaluateFeatureRequest = {
+            requests: request.features.map((feature) => ({
+                entityId,
+                evaluationContext: context,
+                feature
+            }))
+        };
+
+        const httpResponse = await this.dispatchManager.dispatchBatchEvaluateFeature(
+            apiRequest
+        );
+
+        const evaluations: BatchEvaluateFeatureResult = await new Response(
+            httpResponse?.response.body
+        ).json();
+
+        return evaluations.results.reduce((acc: EvaluationResults, cur) => {
+            const evaluation: EvaluationResult = {
+                // Features can be returned as ARNs, only keep the feature name
+                feature: cur.feature.substring(
+                    cur.feature.lastIndexOf('/') + 1
+                ),
+                reason: cur.reason,
+                value: cur.value,
+                variation: cur.variation
+            };
+            return { ...acc, [evaluation.feature]: evaluation };
+        }, {});
+    }
+
+    private useSessionStorage() {
+        return navigator.cookieEnabled && this.config.allowCookies;
```

Comment:
Sorry I missed this before: I don't think that session storage and cookies are the same thing: https://dev.to/iggredible/cookies-vs-local-storage-vs-session-storage-3gp3. This means that a customer who turns off cookies doesn't necessarily want to turn off session storage/caching, right? I think the right check is to just try/fail session storage? https://stackoverflow.com/questions/58152236/what-to-do-when-sessionstorage-access-denied-due-to-browser-settings

Reply:
I originally proposed having a separate config argument for enabling/disabling session storage and we opted instead to reuse allowCookies.

The more I've thought about it though I think it does make sense to separate them and I think having a default always on makes sense for P0. Will be easier to add a parameter in the future if it's an ask. Made the change.